rio-0.1.9.0 broke etc and Stack

The only other affected package is `stack2nix`, I will create a PR disabling this version. This could be undone when https://github.com/commercialhaskell/stack/pull/4725 and https://github.com/roman/Haskell-etc/pull/67 will get merged.

As @qrilka said, it did not fix all of the problems, so I did end up using the constraints.

Here's a snippet of the errors I saw:

```
[ 60 of 127] Compiling Stack.Types.Config ( src/Stack/Types/Config.hs, dist/build/Stack/Types/Config.o )

src/Stack/Types/Config.hs:893:35: error:
    • Could not deduce (MonadFail m) arising from a use of ‘fail’
      from the context: Monad m
        bound by the type signature for:
                   handleExplicitSetupDep :: forall (m :: * -> *).
                                             Monad m =>
                                             (Text, Bool) -> m (Maybe PackageName, Bool)
        at src/Stack/Types/Config.hs:887:5-84
      Possible fix:
        add (MonadFail m) to the context of
          the type signature for:
            handleExplicitSetupDep :: forall (m :: * -> *).
                                      Monad m =>
                                      (Text, Bool) -> m (Maybe PackageName, Bool)
    • In the expression: fail $ show e
      In a case alternative: Left e -> fail $ show e
      In the expression:
        case parsePackageNameFromString $ T.unpack name' of
          Left e -> fail $ show e
          Right x -> return $ Just x
    |
893 |                         Left e -> fail $ show e
    |                                   ^^^^^^^^^^^^^


[ 4 of 13] Compiling System.Etc.Internal.Spec.Parser ( src/System/Etc/Internal
/Spec/Parser.hs, dist/build/System/Etc/Internal/Spec/Parser.o )

src/System/Etc/Internal/Spec/Parser.hs:72:10: error:
    Ambiguous occurrence ‘fail’
    It could refer to either ‘RIO.fail’,
                             imported from ‘RIO’ at src/System/Etc/Internal/Sp
ec/Parser.hs:7:1-10
                             (and originally defined in ‘Control.Monad.Fail’)
                          or ‘Prelude.fail’,
                             imported from ‘Prelude’ at src/System/Etc/Interna
l/Spec/Parser.hs:12:17-20
                             (and originally defined in ‘GHC.Base’)
   |
72 |     then fail "'option' field input requires either 'long' or 'short' set
tings"
   |          ^^^^
```

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -4545,9 +4545,6 @@ packages:
         # https://github.com/commercialhaskell/stackage/issues/4469
         - cmark-gfm < 0.2.0
 
-        # https://github.com/commercialhaskell/stackage/issues/4479
-        - rio < 0.1.9.0
-
         # https://github.com/commercialhaskell/stackage/issues/4483
         - http-media < 0.8.0.0
 

